### PR TITLE
ddl: Wrap TiFlash and its pd rule management into InfoSyncer

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: ref #29924

Problem Summary:
This PR is the second PR to fix problems described in #29924.
This PR wraps TiFlash and it pd rule related operations into a new `TiFlashPlacementManager` in InfoSyncer.
This PR does not implements basic function of TiFlash replica management, which will be introduced in later PRs.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Wrap TiFlash and its pd rule management into InfoSyncer
```

### CODE REVIEW

File: domain/infosync/info.go [hunk 1]
```diff
@@ -925,3 +965,102 @@ func GetLabelRules(ctx context.Context, ruleIDs []string) (map[string]*label.Rul
 	}
 	return is.labelRuleManager.GetLabelRules(ctx, ruleIDs)
 }
+
+// SetPlacementRule is a helper function to set placement rule.
+func SetPlacementRule(ctx context.Context, rule placement.TiFlashRule) error {
```
[the remark this answers is not in the document]
Comment (lcwangchao) [reply domain/infosync/info.go:970]: REST LGTM

[the remark this answers is not in the document]
Comment (CalvinNeo) [reply domain/infosync/info.go:970]: @lcwangchao Have discussed about this with @xhebox, and the conclusion is no for now. I think it has some benefits to keep TiFlash's management independent with PlacementManager for now.

[the remark this answers is not in the document]
Comment (lcwangchao) [reply domain/infosync/info.go]: OK

**Review by lcwangchao — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**